core/src/main/java/io/grpc/transport/ServerListener.java

```diff
@@ -44,5 +43,12 @@
    * @param transport the new transport to be observed.
    * @return a listener for stream creation events on the transport.
    */
-  ServerTransportListener transportCreated(Service transport);
+  ServerTransportListener transportCreated(ServerTransport transport);
+
+  /**
+   * The server is shutting down. No new transports will be processed, but existing streams may
+   * continue. Shutdown is only caused by a call to {@link Server#shutdown()}. All resources have
+   * been released.
+   */
+  void serverShutdown();
```

Comment:
This is using the same terminology used in ExecutorService, Guava's Service, and ClientTransport.Listener. Shutdown() causes an almost immediate shutdown, which means no new things can happen but existing things may still run toward completion. When things are no longer running, the state is "terminated". It would be a more clear if serverTerminated() was present, but I removed it because it wasn't useful at this point.